### ISSUE TITLE
Need better descriptions for Vehicle A and B passability behaviours (and possibly new behaviours)

### ISSUE THREAD

**Issue by ohrrpgce-bugbot** [ohrrpgce/ohrrpgce#187]
[bz#187]

Shouldn't a vehicle set to "Ignore Walls: B" only ignore the walls that are
actually on the B tile, instead of the way it currently works?

Basically, if a vehicle is set to "Ignore Walls: B" , then it won't actually
ignore the walls on tiles set to Vehicle B --- instead, it will ignore ALL
walls, if it's ON a tile set to Vehicle B.

What this means is something set to "Ignore Walls: Not A" , and your A-type tile
is a mountain tile (with full walls), then this vehicle will be able to move
anywhere on the map, including the mountain tiles... but if it enters a mountain
tile, it will then get stuck.

I was almost certain that Ignore Walls was only supposed to work on walls
actually specified on that tile...

Another small problem is that Ignore Walls supercedes "Blocked By" , when I
think it ought not to...  (Why bother having "Blocked By" if it isn't actually
blocked by it if you set "Ignore Walls: everywhere" ?  It'd be nice to have
something that's unblocked by everything BUT A or B-type tiles, but there's no
way to do that as it stands.)

It'd be nice to have it work the way I think it should, since then I could
actually make a hot air balloon that's blocked by mountains and not anything
else.  As it is, there's no way to do that besides plotscripting entire map pass
block changes, and that simply isn't feasible on large, 316x101 maps, for example.

Of course, backwards compatability makes it nasty, but perhaps just add an extra
option: "Ignore walls on tiles: A/B/NotA/NotB/AorB/etc" along with the other
"Ignore walls while on tiles:" option.

From: **JC &lt;diachuk&gt;**
Operating system:   Windows XP
Severity:     feature request

**Comment by ohrrpgce-bugbot** [comment]
Comment author: @bob-the-hamster

Vehicle A and B tiles currently work as intended. The behaviour of "Ignore Walls A" and "ignore Walls B" definitely cannot change-- that would be a serious backwards-compatability break.

I agree that the naming is kinda confusing. Perhaps "Ignore Walls: A" should be renamed to "when on A tiles, ignore walls" or similar.

As for making vehicle behaviour more extensible, we are probably best off thinking about adding C, D, E, etc bits for tiles, rather than adding additional complex and confusinly named A/B tile behaviours

At any rate, this is low priority right now.

**Comment by rversteegen** [comment]
Duplicate of, and better covered by #1144.
